File: frontend/src/types/modelFamily.ts
export interface ModelFamily {
  id: number;
  name: string;
  display_name: string;
  status: "active" | "disabled";
  description: string;
  created_at: string;
  updated_at: string;
}

export interface ModelFamilyInput {
  name: string;
  display_name: string;
  status: ModelFamily["status"];
  description: string;
}

File: backend/internal/entity/model_family.go
package entity

import "time"

type ModelFamily struct {
	ID          int64     `gorm:"primaryKey" json:"id"`
	Name        string    `json:"name"`
	DisplayName string    `json:"display_name"`
	Status      string    `json:"status"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
